### PR TITLE
Remove pickle.loads in averager

### PR DESCRIPTION
**Pull request by justheuristic**
Addresses #155 : require pre-serialized (bytes) metadata in get_current_state, switch to MSGPackSerializer in the application code

### CODE REVIEW

File: hivemind/client/averaging/__init__.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ class DecentralizedAverager(mp.Process, averaging_pb2_grpc.DecentralizedAveragin
     :param dht: a DHT node that will be used to find groups
     :param start: if True, starts the background process immediately
 
-    :param prefix: a shared prefix for all group keys
+    :param prefix: a shared prefix for all group keysP
```
Comment (borzunov) [inline hivemind/client/averaging/__init__.py:45]: Nit: accidental change?

Comment (borzunov) [inline hivemind/client/averaging/__init__.py:45]: Nit: accidental change?

**Review by borzunov — APPROVED** [review]

File: hivemind/client/averaging/__init__.py [hunk 2]
```diff
@@ -376,7 +384,7 @@ async def _load_state_from_peers(self, future: MPFuture):
                     current_tensor_parts, tensors = [], []
                     async for message in stream:
                         if message.metadata:
-                            metadata = PickleSerializer.loads(message.metadata)
```
Comment (mryab) [inline hivemind/client/averaging/__init__.py:379]: PickleSerializer is no longer required, so the import can be removed

File: hivemind/client/averaging/__init__.py [hunk 3]
```diff
@@ -348,8 +348,16 @@ def _background_thread_fetch_current_state_if_asked(self):
...
                 continue
 
-    def load_state_from_peers(self, wait=True) -> Optional[Any]:
-        """ Try to download the latest optimizer state one of the existing peer """
+    def load_state_from_peers(self, wait=True) -> Optional[Tuple[bytes, Sequence[torch.Tensor]]]:
+        """
+        Try to download the latest optimizer state one of the existing peer.
+        :returns: on success, return a 2-tuple with (serialized_metadata, tensors), where
+
+        - serialized_metadata is a small bytestring containing **serialized** metadata (e.g. hyperparameters)
```
Comment (mryab) [inline hivemind/client/averaging/__init__.py:356]: Shouldn't we also deserialize on return? What are the cases where we still need serialized metadata as the output of this method?

**Review by mryab — CHANGES_REQUESTED** [review]

File: hivemind/client/averaging/__init__.py (hunk 3 above)
Comment (justheuristic) [reply hivemind/client/averaging/__init__.py:356]: As of now, this is left to the user (to provide choose a secure pickler).

File: tests/test_averaging.py [hunk 4]
```diff
@@ -277,7 +278,7 @@ def get_current_state(self):
             """
             nonlocal num_calls, super_metadata, super_tensors
             num_calls += 1
-            return super_metadata, super_tensors
+            return serializer.dumps(super_metadata), super_tensors
```
Comment (mryab) [inline tests/test_averaging.py:281]: I believe you can use self.serializer here and avoid declaring the serializer at L271?

**Review by mryab — COMMENTED** [review]
Also bump the version, please

**Review by mryab — APPROVED** [review]

**Pull request merged**